src/tree/azure/AzureResourceTreeDataProvider.ts

```diff
@@ -65,75 +66,87 @@ export class AzureResourceTreeDataProvider extends AzureResourceTreeDataProvider
         if (element) {
             return await element.getChildren();
         } else {
-            const subscriptionProvider = await this.getAzureSubscriptionProvider();
+            const subscriptionProvider = await getAzureSubscriptionProvider(this);
+            const children: ResourceGroupsItem[] = await OnGetChildrenBase(subscriptionProvider, this);
 
-            if (subscriptionProvider) {
-                if (isLoggingIn()) {
-                    return [new GenericItem(
-                        localize('signingIn', 'Waiting for Azure sign-in...'),
-                        {
-                            commandId: 'azureResourceGroups.logIn',
-                            iconPath: new vscode.ThemeIcon('loading~spin')
-                        }
-                    )];
-                } else if (await subscriptionProvider.isSignedIn()) {
-                    this.sendSubscriptionTelemetryIfNeeded();
-                    let subscriptions: AzureSubscription[];
-                    await vscode.commands.executeCommand('setContext', 'azureResourceGroups.needsTenantAuth', false);
-                    if ((subscriptions = await subscriptionProvider.getSubscriptions(true)).length === 0) {
-                        if (
-                            // If there are no subscriptions at all (ignoring filters) AND if unauthenicated tenants exist
-                            (await subscriptionProvider.getSubscriptions(false)).length === 0 &&
-                            (await getUnauthenticatedTenants(subscriptionProvider)).length > 0
-                        ) {
-                            // Subscriptions might exist in an unauthenticated tenant. Show welcome view.
-                            await vscode.commands.executeCommand('setContext', 'azureResourceGroups.needsTenantAuth', true);
-                            return [];
-                        } else {
-                            return [new GenericItem(localize('noSubscriptions', 'Select Subscriptions...'), {
-                                commandId: 'azureResourceGroups.selectSubscriptions'
-                            })]
-                        }
+            if (children.length === 0) {
```

Comment:
For code reading purposes, it might be nice to note here in what cases `OnGetChildrenBase` will return no children. Aka when the user is signed in.